op-node/rollup/clsync/clsync.go

```diff
@@ -117,7 +134,45 @@ func (eq *CLSync) Proceed(ctx context.Context) error {
 		eq.unsafePayloads.Pop()
 		return err
 	}
+
+	if eq.publishAttributes {
+		err = eq.PublishAttributes(ctx, ref)
+		if err != nil {
+			eq.log.Warn("Error publishing L2 attributes", "err", err)
+		}
+	}
+
 	eq.unsafePayloads.Pop()
 	eq.log.Trace("Executed unsafe payload", "hash", ref.Hash, "number", ref.Number, "timestamp", ref.Time, "l1Origin", ref.L1Origin)
 	return nil
 }
+
+func (eq *CLSync) PublishAttributes(ctx context.Context, l2head eth.L2BlockRef) error {
+	l1Origin, err := eq.l1OriginSelector.FindL1Origin(ctx, l2head)
+	if err != nil {
+		eq.log.Error("Error finding next L1 Origin", "err", err)
+		return err
+	}
+
+	fetchCtx, cancel := context.WithTimeout(ctx, time.Millisecond*500)
+	defer cancel()
+
+	attrs, err := eq.attrBuilder.PreparePayloadAttributes(fetchCtx, l2head, l1Origin.ID())
+	if err != nil {
+		eq.log.Error("Error preparing payload attributes", "err", err)
+		return err
+	}
+
+	withParent := &derive.AttributesWithParent{
+		Attributes:   attrs,
+		Parent:       l2head,
+		IsLastInSpan: false,
```

Comment:
what is `IsLastInSpan` used for?

Reply:
the derivation pipeline inserts blocks in batch which they call it a span. If it is the last block in span, it sets that to the safe head.
https://github.com/flashbots/optimism/blob/f8143c8cbc4cc0c83922c53f17a1e47280673485/op-node/rollup/attributes/attributes.go#L133

Reply:
since we are inserting unsafe payloads it should be set to `false`